docs/proposals/20210203-unmanaged-infrastructure.md

```diff
@@ -0,0 +1,226 @@
+  - "@CecileRobertMichon"
+  - "@yastij"
+  
+creation-date: 2021-02-03
+last-updated: 2021-02-12
+status: implementable
+see-also:
+replaces:
+superseded-by:
+---
+
+# Externally Managed cluster infrastucture
+
+## Table of Contents
+
+A table of contents is helpful for quickly jumping to sections of a proposal and for highlighting
+any additional information provided beyond the standard proposal template.
+[Tools for generating](https://github.com/ekalinin/github-markdown-toc) a table of contents from markdown are available.
+
+- [Title](#title)
+  - [Table of Contents](#table-of-contents)
+  - [Glossary](#glossary)
+  - [Summary](#summary)
+  - [Motivation](#motivation)
+    - [Goals](#goals)
+    - [Non-Goals/Future Work](#non-goalsfuture-work)
+  - [Proposal](#proposal)
+    - [User Stories](#user-stories)
+      - [Story 1](#story-1)
+      - [Story 2](#story-2)
+    - [Requirements (Optional)](#requirements-optional)
+      - [Functional Requirements](#functional-requirements)
+        - [FR1](#fr1)
+        - [FR2](#fr2)
+      - [Non-Functional Requirements](#non-functional-requirements)
+        - [NFR1](#nfr1)
+        - [NFR2](#nfr2)
+    - [Implementation Details/Notes/Constraints](#implementation-detailsnotesconstraints)
+    - [Security Model](#security-model)
+    - [Risks and Mitigations](#risks-and-mitigations)
+  - [Alternatives](#alternatives)
+  - [Upgrade Strategy](#upgrade-strategy)
+  - [Additional Details](#additional-details)
+    - [Test Plan [optional]](#test-plan-optional)
+    - [Graduation Criteria [optional]](#graduation-criteria-optional)
+    - [Version Skew Strategy [optional]](#version-skew-strategy-optional)
+  - [Implementation History](#implementation-history)
+
+## Glossary
+
+Refer to the [Cluster API Book Glossary](https://cluster-api.sigs.k8s.io/reference/glossary.html).
+
+### Managed cluster infrastructure
+
+Cluster infratructure which lifecycle is managed by a provider infraCluster CR.
+E.g in AWS:
+- Network
+    - VPC
+    - Subnets
+    - Internet gateways
+    - Nat gateways
+    - Route tables
+- Security groups
+- Load balancers
+
+### Externally managed cluster infrastructure
+Cluster infratructure which lifecycle is not managed by CAPI but rather by an external entity.
+
+## Summary
+
+This proposal introduces first class support for "externally managed" infrastructure for CAPI providers.
+This consolidates the boundaries between CAPI managed (existing pattern) and externally managed ("Bring Your Own") cluster infrastructure.
+
+## Motivation
+
+Currently, Cluster API infrastructure providers support an opinionated happy path to create and manage cluster infrastructure lifecycle.
+The fundamental use case we want to support is out of tree controllers or tools that can manage this cluster infrastructure.
+
+For example, this could allow users to create clusters using tooling such as Terraform or Kops and allow them to add CAPI Machine infrastructure as a day 2 operation.
```

Comment:
```suggestion
For example, users could create clusters using tools such as Terraform, Crossplane, or Kops.
```

What does "CAPI Machine infrastructure as a day 2 operation" mean in this context?

Reply:
I believe this means using CAPI on infrastructure that was already configured and installed.